### PR TITLE
Python SDK: crud tests for looks

### PR DESCRIPTION
**Pull request by josephaxisa**
- Added a test that creates a test look using data.yml
- A fixture that deletes the test look before the crud tests run

### CODE REVIEW

File: python/tests/integration/test_methods.py [hunk 1]
```diff
@@ -254,9 +245,47 @@ def test_it_runs_inline_query(sdk: mtds.LookerSDK, queries: TQueries):
...
+                deleted=l.get("deleted"),
+                is_run_on_load=l.get("is_run_on_load"),
+                public=l.get("public"),
+                query_id=query.id,
+                space_id=l.get("space_id") or str(sdk.me().personal_space_id),
+            )
+        )
+
+        assert isinstance(look, ml.LookWithQuery)
+        assert look.title == l.get("title")
```
Comment (jkaster) [inline python/tests/integration/test_methods.py]: Is `obj.get("...")` the reference syntax for accessing properties of an object? That seems to be an anti-pattern

File: python/tests/conftest.py [hunk 2]
```diff
@@ -53,6 +58,15 @@ def remove_test_dashboards(sdk: mtds.LookerSDK, dashboards):
                 sdk.delete_dashboard(cast(str, dashboard.id))
 
 
+@pytest.fixture(scope="function")
+def remove_test_looks(sdk: mtds.LookerSDK, looks):
+    for l in looks:
+        search_results = sdk.search_looks(title=l["title"])
+        if len(search_results) > 0:
+            for look in search_results:
+                sdk.delete_look(cast(int, look.id))
```
Comment (jkaster) [inline python/tests/conftest.py]: If we add an additional look to data.yml, will this loop actually work to match all looks and delete them?

File: python/tests/integration/test_methods.py (hunk 1 above)
Comment (josephaxisa) [reply python/tests/integration/test_methods.py]: `obj.get("property")` returns `None` if no second argument is provided. If we did `obj["property"]` we'd get a `KeyError` when the property doesn't exist

Comment (josephaxisa) [reply python/tests/integration/test_methods.py]: At the same time, some of these properties are required so perhaps they should be accessed using the latter

File: python/tests/conftest.py (hunk 2 above)
Comment (josephaxisa) [reply python/tests/conftest.py]: Yes. `looks` on line 62 is a list containing all looks defined in `data.yml`. As long as each look has a title, we'll be able to find it and delete it.

File: python/tests/integration/test_methods.py (hunk 1 above)
Comment (jkaster) [reply python/tests/integration/test_methods.py]: good note about the accessor error. I didn't consider that. I think for the required fields we should use `foo.bar` notation unless it becomes a hassle to check the property status.

File: python/tests/conftest.py (hunk 2 above)
Comment (jkaster) [reply python/tests/conftest.py]: sorry, I missed the iterator on line 63!

File: python/tests/integration/test_methods.py (hunk 1 above)
Comment (josephaxisa) [reply python/tests/integration/test_methods.py]: fwiw, we used the `.get` notation everywhere else throughout the tests.  In this case, `looks` is a list of dictionaries. The notation we can use is either `obj[key]` or `obj.get(key[, default])` (https://docs.python.org/3/library/stdtypes.html?highlight=dict%20get#dict.get). It wouldn't introduce any hassle to use `obj[key]` but I might need to go back and change all the other tests for consistency if we're going down this route. Based on the docs the only required parameters for `create_look` are `model` and `view` in `query`.

Comment (jkaster) [reply python/tests/integration/test_methods.py]: I think `obj[key]` is slightly better than `obj.get(key...)` unless the semantics of failed gets changes

File: python/tests/integration/test_methods.py [hunk 3]
```diff
@@ -254,9 +245,47 @@ def test_it_runs_inline_query(sdk: mtds.LookerSDK, queries: TQueries):
...
+        assert look.description == l.get("description")
+        assert look.deleted == l.get("deleted", False)
+        assert look.is_run_on_load == l.get("is_run_on_load", True)
+        assert look.public == l.get("public", False)
```
[the remark this answers is not in the document]
Comment (jkaster) [reply python/tests/integration/test_methods.py]: I like making the yaml more explicit

File: python/tests/integration/test_methods.py (hunk 1 above)
Comment (jkaster) [reply python/tests/integration/test_methods.py]: Separate PR FTW

Comment (josephaxisa) [reply python/tests/integration/test_methods.py]: Perfect, agreed. I'll leave these as is for now and create a separate PR first chance I get.

**Pull request merged**